File: sdks/node/examples/agent_with_lazai.ts
import { Client, ProofData } from "alith/lazai";

async function main() {
  const contributor = new Client();
  const node = new Client();
  console.log("wallet address", contributor.getWallet().address);
  const url = "https://example.com/okyes.txt";
  let fileId = await contributor.getFileIdByUrl(url);
  // File not found, add it
  if (fileId === BigInt(0)) {
    fileId = await contributor.addFile(url);
  }
  console.log("file id:", fileId);
  let nodeInfo = await node.getNode(node.getWallet().address);
  if (nodeInfo.url.length > 0) await node.removeNode(node.getWallet().address);
  await node.addNode(
    node.getWallet().address,
    "https://example.com/node",
    "0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"
  );
  const nodeFee = BigInt(10);
  await node.updateNodeFee(nodeFee);
  await contributor.requestProof(fileId, nodeFee);
  const ids = await contributor.fileJobIds(fileId);
  console.log("file proof job ids", ids);
  const job_id = ids[ids.length - 1];
  const job = await contributor.getJob(job_id);
  console.log("job id", job_id);
  console.log("job", job);
  nodeInfo = await contributor.getNode(job.nodeAddress);
  console.log("node info:", nodeInfo);
  await node.completeJob(job_id);
  console.log("completed job", job_id);
  await node.addProof(fileId, new ProofData(Number(fileId), "", ""));
  console.log("proof added for file", fileId);
  await contributor.requestReward(fileId);
  console.log("reward requested for file", fileId);
}
await main();

File: sdks/python/examples/agent_with_lazai.py
from alith.lazai import Client, ProofData

contributor = Client()
node = Client()
print("wallet address", contributor.wallet.address)
url = "https://example.com/okyes.txt"
file_id = contributor.get_file_id_by_url(url)
# File not found, add it
if file_id == 0:
    # Note: just for demo, we don't upload the file content hash here.
    file_id = contributor.add_file(url)
print("file id", file_id)
node_url = node.get_node(node.wallet.address)[1]
if node_url:
    node.remove_node(node.wallet.address)
node.add_node(
    node.wallet.address,
    "https://example.com/node",
    "0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef",
)
node_fee = 10
node.update_node_fee(node_fee)
contributor.request_proof(file_id, node_fee)
ids = contributor.file_job_ids(file_id)
print("file proof job ids", ids)
job_id = ids[-1]
job = contributor.get_job(job_id)
print("job id", job_id)
print("job", job)
node_info = contributor.get_node(job[-1])
print("node info", node_info)
node.complete_job(job_id)
print("job completed for id", job_id)
node.add_proof(file_id, ProofData(id=file_id, score=1, file_url="", proof_url=""))
print("proof added for file id", file_id)
contributor.request_reward(file_id)
print("reward requested for file id", file_id)
